keps/sig-node/compute-device-assignment.md

```diff
@@ -19,99 +19,73 @@ creation-date: "2018-07-19"
 
 ![device monitoring architecture](https://user-images.githubusercontent.com/3262098/43926483-44331496-9bdf-11e8-82a0-14b47583b103.png)
 
-### Kubelet Device Lifecycle
-
-* As a cluster operator, I want the Kubelet to have consistent behavior during kubelet restarts, node reboots, and node upgrades regardless of any disk failure/corruption or otherwise unrecoverable node state.  Since the Kubelet now obtains container-device bindings from pod objects, the Kubelet's Device Manager no longer needs to checkpoint these bindings or implement complex reasoning about state upon startup.
 
-### Out-of-tree Experimentation
-
-#### Network Device Plugins
-
-* As a _Device Vendor_, I manufacture network devices.  Even though the Kubelet exposes different interfaces for network setup (CNI), and device lifecycle management (Device Plugin), I can enable consumption of my network devices without requiring communication between my CNI and Device Plugin implementations.  This allows me to independently develop, test, and release my component for network setup, and my component for managing network devices without any dependencies between them. My Device Plugin exposes the set of network devices on the node, but does _not_ perform network setup when `Allocate` is called. Instead, when `CNI_ADD` is called on my CNI plugin, it queries the kubelet's `/pods` endpoint to discover the network devices it should perform network setup for:
-
-![network device architecture](https://user-images.githubusercontent.com/3262098/43926528-65444380-9bdf-11e8-9ac2-49c8ae46d3a1.png)
+## Changes
 
-#### Guaranteed Device Scheduling
+Add a Kubelet endpoint, `/devices/<ResourceName>`, wich returns information about the kubelet's assignment of devices to containers.  It returns a single `PodDeviceList`, which is shown below:
+```golang
+// PodDeviceList is a list of PodDevice's
+type PodDeviceList []PodDevices
+
+// PodDevices contains information about the devices assigned to a pod
+type PodDevices struct {
+    PodName       string
+    PodNamespace  string
+    Containers []ContainerDevices
+}
 
-* As a _Cluster Admin and Application Developer_, I have a workload with strict requirements about the set of devices it is assigned that cannot be expressed with current kubernetes APIs (e.g. numa topology requirements). If my workload is assigned to a node that has the devices I need, but not in the correct configuration, my workloads performance suffers considerably.  Thus, the node's best-effort attempt to provide a "good" set of devices is not enough to meet the needs of my workload.  I can run a custom scheduler to assign devices to containers during scheduling, and can _guarantee_ that the set of devices the workload is assigned meets my requirements.
+// ContainerDevices contains information about the devices assigned to a container
+type ContainerDevices struct {
+    ContainerName string
+    DeviceIDs     []string
```

Comment:
Do we want to make this a map keyed by resourceName?

Reply:
since endpoint is `/devices/<ResourceName>`, `DeviceIDs` should belong to only <ResourceName>.

Reply:
Yes.  The assumption here is that callers are only concerned with one resource.  If this assumption doesn't work for some use cases, we could make `/devices` return a `map[ResourceName]PodDevices` for those callers that need to know all devices from all resources.